File: backend/api/admin/a_matches.go
package apiAdmin

import (
	"backend/models"
	"backend/services"
	"encoding/json"
	"net/http"
)

/* ---------------- UPDATE MATCH ---------------- */

func UpdateMatch(w http.ResponseWriter, r *http.Request) {

	if r.Method == http.MethodOptions {
		return
	}
	if r.Method != http.MethodPut {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var req models.Match
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}

	match, err := services.UpdateMatch(req)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(match)
}

/* ---------------- DELETE MATCH ---------------- */

func DeleteMatch(w http.ResponseWriter, r *http.Request) {

	if r.Method == http.MethodOptions {
		return
	}
	if r.Method != http.MethodDelete {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var req struct {
		ID string `json:"id"`
	}
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}

	match, err := services.DeleteMatch(req.ID)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(match)
}

/* ---------------- CREATE MATCH ---------------- */

func CreateNewMatch(w http.ResponseWriter, r *http.Request) {

	if r.Method == http.MethodOptions {
		return
	}
	if r.Method != http.MethodPost {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var req models.Match
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}

	match, err := services.CreateNewMatch(req)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(match)
}

/* ---------------- CHANGE DAY SCHEDULE ---------------- */

func ChangeSchedule(w http.ResponseWriter, r *http.Request) {

	if r.Method == http.MethodOptions {
		return
	}
	if r.Method != http.MethodPut {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var req models.ScheduleDay
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}

	if err := services.ChangeDaySchedule(req); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.WriteHeader(http.StatusOK)
}

File: lib/api/admin/a_matches.ts
import { Match, ScheduleDay } from "@/types/matches";
import { API_BASE_URL } from "@/lib/config";

export async function updateMatch( match: Match): Promise<Match> {
  const response = await fetch (`${API_BASE_URL}/matches/update`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      },
      body: JSON.stringify({
        ID: match.id,
        Date: match.date,
        Time: match.time,
        HomeTeam: match.homeTeam,
        AwayTeam: match.awayTeam,
        Field: match.field,
        HomeScore: match.homeScore,
        AwayScore: match.awayScore,
      }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw new Error(`Failed to edit match: ${text}`);
  }
  return await response.json();
    
}

export async function deleteMatch( match: Match): Promise<Match> {
  const response = await fetch (`${API_BASE_URL}/matches/delete`, {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json",
      },
      body: JSON.stringify({
        ID: match.id,
      }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw new Error(`Failed to delete match: ${text}`);
  }
  return await response.json();
    
}

export async function createNewMatch( match: Match): Promise<Match> {
  const response = await fetch (`${API_BASE_URL}/matches/create`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      },
      body: JSON.stringify({
        Date: match.date,
        Time: match.time,
        HomeTeam: match.homeTeam,
        AwayTeam: match.awayTeam,
        Field: match.field,
        HomeScore: match.homeScore,
        AwayScore: match.awayScore,
      }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw new Error(`Failed to create new match: ${text}`);
  }
  return await response.json();
    
}

export async function changeSchedule( days: ScheduleDay[]):  Promise<void>{
  const response = await fetch (`${API_BASE_URL}/matches/schedule`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      },
      body: JSON.stringify({ days }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw new Error(`Failed to change day schedule: ${text}`);
  }
  return await response.json();
} 